File: dotnet/dojos/dojo1/SecondPractice/BowlingSecondPractice/LastFrame.cs
namespace BowlingSecondPractice
{
    public class LastFrame : Frame
    {
        public LastFrame(int firstBall, int secondBall, int thirdBall) : base(firstBall, secondBall)
        {
            ThirdBall = thirdBall;
        }

        public int ThirdBall { get; }

        protected override int CalculateScore()
        {
            return FirstBall + SecondBall + ThirdBall;
        }
    }
}

File: dotnet/dojos/dojo1/JBG/src/main/java/bowling/seven/LastFrame.java
package bowling.seven;

public class LastFrame extends Frame {
    private final int firstBall;
    private final int secondBall;
    private final int thirdBall;

    public LastFrame(int firstBall, int secondBall, int thirdBall) {
        super(firstBall, secondBall);
        this.firstBall = firstBall;
        this.secondBall = secondBall;
        this.thirdBall = thirdBall;
    }

    @Override
    public int countScore() {
        return firstBall + secondBall + thirdBall;
    }
}
